File: java-example/src/main/java/org/onflow/examples/java/getAccount/GetAccountAccessAPIConnector.java
package org.onflow.examples.java.getAccount;

import org.onflow.flow.sdk.*;
import java.math.BigDecimal;

public class GetAccountAccessAPIConnector {
    private final FlowAccessApi accessAPI;

    public GetAccountAccessAPIConnector(FlowAccessApi accessAPI) {
        this.accessAPI = accessAPI;
    }

    public FlowAccount getAccountAtLatestBlock(FlowAddress address) {
        FlowAccessApi.AccessApiCallResponse<FlowAccount> response = accessAPI.getAccountAtLatestBlock(address);
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowAccount>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }


    public FlowAccount getAccountAtBlockHeight(FlowAddress address, long height) {
        FlowAccessApi.AccessApiCallResponse<FlowAccount> response = accessAPI.getAccountByBlockHeight(address, height);
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowAccount>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }

    public BigDecimal getAccountBalance(FlowAddress address) {
        FlowAccount account = getAccountAtLatestBlock(address);
        return account.getBalance();
    }
}

File: kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/getAccount/GetAccountAccessAPIConnector.kt
package org.onflow.examples.kotlin.getAccount

import org.onflow.flow.sdk.*
import java.math.BigDecimal

internal class GetAccountAccessAPIConnector(
    private val accessAPI: FlowAccessApi
) {
    fun getAccountAtLatestBlock(address: FlowAddress): FlowAccount =
        when (val response = accessAPI.getAccountAtLatestBlock(address)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }

    fun getAccountAtBlockHeight(address: FlowAddress, height: Long): FlowAccount =
        when (val response = accessAPI.getAccountByBlockHeight(address, height)) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }

    fun getAccountBalance(address: FlowAddress): BigDecimal {
        val account = getAccountAtLatestBlock(address)
        return account.balance
    }
}
